### PR TITLE
feat(rust): Format null arrays in Series

### LINKED ISSUE #7153: Format null arrays in Series like other series and format pl.List(pl.Null) arrays properly in dataframes.
### Problem description

Format null arrays in Series like other series and format pl.List(pl.Null) arrays properly in dataframes.

Currently:

  - Formatting of dataframe columns with pl.List(pl.Null) does not work properly.
  - A plain Polars Null series does not get printed like other series.

```python
>>> tbl = pa.table({"a": [None, None], "b": [[None, None], [None, None]]})
>>> pl.from_arrow(tbl)
shape: (2, 2)
┌──────┬─────────────────┐
│ a    ┆ b               │
│ ---  ┆ ---             │
│ null ┆ list[null]      │
╞══════╪═════════════════╡
│ null ┆ fmt implemented │
│ null ┆ fmt implemented │
└──────┴─────────────────┘
>>> df = pl.from_arrow(tbl)
>>> df["a"]
nullarray
>>> pl.Series("a", [None, None], dtype=pl.Null)
nullarray
```

Compared with other (non pl.Null) Series:

```python
>>> tbl_not_null = pa.table({"a": ["None", None], "b": [["None", None], [None, "None"]]})
>>> df = pl.from_arrow(tbl_not_null)
>>> df
shape: (2, 2)
┌──────┬────────────────┐
│ a    ┆ b              │
│ ---  ┆ ---            │
│ str  ┆ list[str]      │
╞══════╪════════════════╡
│ None ┆ ["None", null] │
│ null ┆ [null, "None"] │
└──────┴────────────────┘

>>> df["a"]
shape: (2,)
Series: 'a' [str]
[
        "None"
        null
]

>>> pl.Series("a", [None, "None"], dtype=pl.Utf8)
shape: (2,)
Series: 'a' [str]
[
        null
        "None"
]
```

### PR DESCRIPTION
**Pull request by romanovacca**
Resolves #7153

null array now is shown as:

``` 
shape: (2,)
Series: 'a' [nullarray]
[
        null
        null
] 
``` 

instead of 

```
nullarray
```

Could refactor the code a bit cause it overlaps with `format_array` a lot, so there is quite some repitition, but made this PR to see if this is in the right direction at all.

### CODE REVIEW

**Comment by romanovacca** [comment]
CI failures seem unrelated?

**Comment by romanovacca** [comment]
> Can you rebase based on `main`? There's a lot of stuff in your PR that shouldn't be in there.

Yeah i was fighting with that, never used rebase before. Should be good now!

File: crates/polars-core/src/fmt.rs [hunk 1]
```diff
@@ -313,7 +359,7 @@ impl Debug for Series {
                 "Series"
             ),
             DataType::Null => {
-                writeln!(f, "nullarray; shape: {}", self.len())
+                format_nullarray!(f, self.len(), "nullarray", self.name(), "Series")
```
Comment (c-peters) [inline crates/polars-core/src/fmt.rs:362]: We should be able to reuse the `format_array` macro for the Null Datype. If you downcast the Series into a NullChunked   it should work.

Comment (c-peters) [reply crates/polars-core/src/fmt.rs:362]: We should probably rename `nullarray` to `null` to be more consistent

Comment (romanovacca) [reply crates/polars-core/src/fmt.rs]: I couldn't see a direct way to downcast into NullChunked, so I added some logic in order to do that. Is this the correct way? It works, but not sure if this is the implementation you meant

**Comment by romanovacca** [comment]
```
>>> df_null 
shape: (2, 1)
┌──────┐
│ a    │
│ ---  │
│ null │
╞══════╡
│ null │
│ null │
└──────┘


>>> df_null["a"]
shape: (2,)
Series: 'a' [null]
[
        null
        null
]
>>> 
```

**Review by ritchie46 — APPROVED** [review]

**Comment by ritchie46** [comment]
Thank you @romanovacca. Looks better indeed. :)

**Pull request merged**